.github/workflows/build.yml

```diff
@@ -111,12 +119,25 @@ jobs:
           PR_URL: ${{github.event.pull_request.html_url}}
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
   canary-release:
-    if: github.event_name != 'pull_request' && !startsWith(github.ref, 'refs/tags/v2')
+    if: ${{ github.event_name == 'workflow_dispatch' && inputs.canary-release == true || github.event_name == 'schedule' }}
     runs-on: ubuntu-latest
     needs: [tests, checks, e2e-tests]
     steps:
       - uses: actions/checkout@v3
       - run: git fetch --depth=1
+      - name: Check if latest commit is within 24 hrs
+        run: |
+          lastCommitDate=$(git --no-pager log -n 1 main --pretty=format:"%at")
+          curDate=$(date +%s)
+          dateDiff=$(expr $curDate - $lastCommitDate)
+          echo $lastCommitDate, $curDate, $dateDiff
+
+          if [[ $dateDiff -gt 86400 ]]
+          then
+              echo 'No new commit found on ${{ github.ref }} within the last 24 hrs.'
+              echo 'Canceling the canary release...'
+              curl -X POST -H "Accept: application/vnd.github+json" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/cancel
```

Comment:
[q] isn't there a more elegant way to cancel the current run instead of calling the api from a running job?